org.ekstep.summary-1.0/renderer/plugin.js

```diff
@@ -23,39 +23,10 @@ org.ekstep.summaryRenderer = Plugin.extend({ // eslint-disable-line no-undef
     var instance = this;
     var assessTelemetryData = org.ekstep.service.content.getTelemetryEvents();
     _.forEach(assessTelemetryData.assess, function(value,key) {
-      var item = value.edata.item;
-      switch(item.type){
-        case 'ftb': if(_.isEmpty(value.edata.resvalues)){
-                      instance.setNonAttemptedQuestion(key);                        
-                    }else{
-                      instance.setAttemptedQuestion(key);
-                    }
-                    break;
-        case 'mcq': if(_.isEmpty(value.edata.resvalues[0])){
-                      instance.setNonAttemptedQuestion(key);
-                    }else{
-                      instance.setAttemptedQuestion(key);
-                    }
-                    break;
-        case 'reorder': if(_.isEmpty(value.edata.resvalues)){
-                      instance.setNonAttemptedQuestion(key);
-                    }else{
-                      instance.setAttemptedQuestion(key);
-                    }
-                    break;
-        case 'mtf': if(_.isEqual(item.params[1], value.edata.resvalues[1])){
-                      instance.setNonAttemptedQuestion(key);                        
-                    }else{
-                      instance.setAttemptedQuestion(key);
-                    }
-                    break;
-        case 'sequence': item.params.pop(); 
-                      if(_.isEqual(item.params, value.edata.resvalues)){
-                        instance.setNonAttemptedQuestion(key);                        
-                      }else{
-                        instance.setAttemptedQuestion(key);
-                      }
-                      break;
+      if(_.isEmpty(value.edata.resvalues)){
```

Comment:
Good work.
Also, merge both setNonAttemptedQuestion & setAttemptedQuestion into single function.
attempted-> boolean value(attempted or not)

setNonAttemptedQuestion: function(questionId, attempted){
    var instance = this;
    var attenmpKey = attempted ? 'ttempted' : nonAttempted;
    if(!_.includes(instance._qsSummary[attenmpKey], questionId))
      instance._qsSummary[attenmpKey].push(questionId);
      var index = instance._qsSummary[attenmpKey].indexOf(questionId);
      if (index !== -1) instance._qsSummary[attenmpKey].splice(index, 1);
  },

Reply:
@vinukumar-vs Question attempted or not we are deciding based on Assess event resValues so we cant merge both methods. Since both methods meant for their specific task
@rahulshukla